lib/zclient.c

```diff
@@ -143,6 +143,18 @@ void redist_del_instance(struct redist_proto *red, unsigned short instance)
 	}
 }
 
+void redist_del_all_instances(struct redist_proto *red)
+{
+	struct listnode *ln, *nn;
+	unsigned short *id;
+
+	if (!red->instances)
```

Comment:
isn't this check done in `ALL_LIST_ELEMENTS`?

Reply:
I guess it is, by way of using listhead? still prefer the explicit check honestly